lib/KangarooTwelve.c

```diff
@@ -21,16 +21,15 @@ and related or neighboring rights to the source code in this file.
 
 /* ---------------------------------------------------------------- */
 
+// TODO: compute the values dynamically, based on `instance->rate`
 #define K12_security        128
 #define K12_capacity        (2*K12_security)
 #define K12_capacityInBytes (K12_capacity/8)
-#define K12_rate            (1600-K12_capacity)
-#define K12_rateInBytes     (K12_rate/8)
-#define K12_rateInLanes     (K12_rate/64)
 
-static void TurboSHAKE128_Initialize(TurboSHAKE128_Instance *instance)
+static void TurboSHAKE128_Initialize(TurboSHAKE128_Instance *instance, int securityLevel)
```

Comment:
OK to add a security strength level parameter, but then the function name cannot be TurboSHAKE128 anymore.